tests/crafting_test.cpp

```diff
@@ -425,6 +425,29 @@ static int actually_test_craft( const recipe_id &rid, const std::vector<item> to
     return turns;
 }
 
+// Resume the first in progress craft found in the player's inventory
+static int resume_craft()
+{
+    item *craft = g->u.items_with( []( const item & itm ) {
+        return itm.is_craft();
+    } ).front();
+    const recipe &rec = craft->get_making();
+    set_time( midday ); // Ensure light for crafting
+    REQUIRE( g->u.morale_crafting_speed_multiplier( rec ) == 1.0 );
+    REQUIRE( g->u.lighting_craft_speed_multiplier( rec ) == 1.0 );
+    REQUIRE( !g->u.activity );
+    g->u.use( g->u.get_item_position( craft ) );
+    CHECK( g->u.activity );
+    CHECK( g->u.activity.id() == activity_id( "ACT_CRAFT" ) );
+    int turns = 0;
+    while( g->u.activity.id() == activity_id( "ACT_CRAFT" ) ) {
```

Comment:
You seem to be hanging here.

Reply:
Yeah, I forgot a cancel_activity() call in b544a327ec7e26aed9e0c048e3ac2039bb10258a.  Should be fixed in 7226ec1